### PR TITLE
Add data source "google_privileged_access_manger_entitlement"

### PR DESCRIPTION
**Pull request by varsharmavs**
```release-note:new-datasource
`google_privileged_access_manager_entitlement`
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/privilegedaccessmanager/data_source_privileged_access_manager_entitlement_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	})
+}
+
+func testAccDataSourceGooglePrivilegedAccessManagerEntitlement_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_privileged_access_manager_entitlement" "tfentitlement" {
+	entitlement_id = "tf-test-example-entitlement%{random_suffix}"
+	location = "global"
+	max_request_duration = "43200s"
+	parent = "projects/%{project_name}"
```
Comment (melinath) [inline mmv1/third_party/terraform/services/privilegedaccessmanager/data_source_privileged_access_manager_entitlement_test.go:61]: `project_name` is missing from the context map in both of these test configurations. If the data source doesn't default to a project id for the parent, you'll need to use a `google_project` datasource to get the test project's project_id and interpolate it here. (Search for `data "google_project"` in the codebase for examples.)

**Review by melinath — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/privilegedaccessmanager/data_source_privileged_access_manager_entitlement.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGooglePrivilegedAccessManagerEntitlement() *schema.Resource {
```
Comment (melinath) [inline mmv1/third_party/terraform/services/privilegedaccessmanager/data_source_privileged_access_manager_entitlement.go:11]: you'll need to also add this to handwrittenDatasources in magic-modules/mmv1/third_party/terraform/provider/provider_mmv1_resources.go.erb - sorry for not catching this last time!

File: mmv1/third_party/terraform/services/privilegedaccessmanager/data_source_privileged_access_manager_entitlement_test.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	})
+}
+
+func testAccDataSourceGooglePrivilegedAccessManagerEntitlement_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_privileged_access_manager_entitlement" "tfentitlement" {
+	entitlement_id = "tf-test-example-entitlement%{random_suffix}"
+	location = "global"
+	max_request_duration = "43200s"
+	parent = "projects/${data.google_project.project.number}"
```
Comment (melinath) [inline mmv1/third_party/terraform/services/privilegedaccessmanager/data_source_privileged_access_manager_entitlement_test.go:61]: This is causing tests to fail with:

```
Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 6, in resource "google_privileged_access_manager_entitlement" "tfentitlement":
           6: 	parent = "projects/${data.google_project.project.number}"
```

Possibly the dependencies are order-sensitive, and the datasource needs to go at the top of the test config?

**Review by melinath — DISMISSED** [review]

File: mmv1/third_party/terraform/provider/provider_mmv1_resources.go.erb [hunk 4]
```diff
@@ -170,6 +170,7 @@ var handwrittenDatasources = map[string]*schema.Resource{
 	"google_netblock_ip_ranges":                        resourcemanager.DataSourceGoogleNetblockIpRanges(),
 	"google_organization":                              resourcemanager.DataSourceGoogleOrganization(),
 	"google_privateca_certificate_authority":           privateca.DataSourcePrivatecaCertificateAuthority(),
+	"google_privileged_access_manager_entitlement":		privilegedaccessmanager.DataSourceGooglePrivilegedAccessManagerEntitlement(),
```
Comment (NickElliot) [inline mmv1/third_party/terraform/provider/provider_mmv1_resources.go.erb:173]:
```suggestion
	"google_privateca_certificate_authority":           privateca.DataSourcePrivatecaCertificateAuthority(),
	"google_privileged_access_manager_entitlement":     privilegedaccessmanager.DataSourceGooglePrivilegedAccessManagerEntitlement(),
```

**Review by NickElliot — APPROVED** [review]
Aligns with the PAM Entitlement resource implemented previously and tests are good, LGTM! One comment for formatting that I'll commit in myself so this can be merged before go conversion.

**Pull request merged**